sdk/src/metrics/metric_reader.cc

```diff
@@ -36,7 +36,9 @@ bool MetricReader::Collect(
     OTEL_INTERNAL_LOG_WARN("MetricReader::Collect invoked while Shutdown in progress!");
   }
 
-  return metric_producer_->Collect(callback);
+  auto result  = metric_producer_->Produce();
+  auto success = callback(result.points_);
```

Comment:
This can invoke callback on failures.

Per the spec:

> When the Produce operation fails, the MetricProducer MAY return successfully collected results and a failed reasons list to the caller.

If calling callback on the part that was successfully collected is intended, please add  a comment that explains this, otherwise the code just looks broken at first read.